custom/icds_reports/utils/data_pull.sql

```diff
@@ -0,0 +1,203 @@
+--                                        ->  Parallel Append  (cost=0.00..73930.32 rows=16362 width=4)
+--                                              ->  Parallel Seq Scan on "child_health_monthly_2020-05-01_671996" child_health_monthly  (cost=0.00..73848.51 rows=16362 width=4)
+--                                                    Filter: ((month = '2020-05-01'::date) AND (valid_in_month = 1) AND (state_id = 'f9b47ea2ee2d8a02acddeeb491d3e175'::text) AND ((age_tranche)::integer <= 36))
+
+SELECT caste, count(*) as number_3_6_child FROM "child_health_monthly" WHERE
+    month='2020-05-01'
+    AND age_tranche::Integer>36 AND age_tranche::Integer<=72
+    AND valid_in_month=1
+    AND state_id='f9b47ea2ee2d8a02acddeeb491d3e175'
+    GROUP BY caste;
+
+--                                                                                                                     QUERY PLAN
+-- --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+--  HashAggregate  (cost=0.00..0.00 rows=0 width=0)
+--    Group Key: remote_scan.caste
+--    ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+--          Task Count: 64
+--          Tasks Shown: One of 64
+--          ->  Task
+--                Node: host=100.71.184.232 port=6432 dbname=icds_ucr
+--                ->  Finalize GroupAggregate  (cost=76594.02..76699.15 rows=200 width=12)
+--                      Group Key: child_health_monthly.caste
+--                      ->  Gather Merge  (cost=76594.02..76693.15 rows=800 width=12)
+--                            Workers Planned: 4
+--                            ->  Partial GroupAggregate  (cost=75593.96..75597.80 rows=200 width=12)
+--                                  Group Key: child_health_monthly.caste
+--                                  ->  Sort  (cost=75593.96..75594.57 rows=246 width=4)
+--                                        Sort Key: child_health_monthly.caste
+--                                        ->  Parallel Append  (cost=0.00..75584.19 rows=246 width=4)
+--                                              ->  Parallel Seq Scan on "child_health_monthly_2020-05-01_671996" child_health_monthly  (cost=0.00..75582.96 rows=246 width=4)
+--                                                    Filter: ((month = '2020-05-01'::date) AND (valid_in_month = 1) AND (state_id = 'f9b47ea2ee2d8a02acddeeb491d3e175'::text) AND ((age_tranche)::integer > 36) AND ((age_tranche)::integer <= 72))
+-- (18 rows)
+
+SELECT ccs.caste, count(*) as number_pw FROM "ccs_record_monthly" ccs
+    LEFT JOIN "awc_location" awc ON (awc.doc_id = ccs.awc_id AND awc.supervisor_id = ccs.supervisor_id)
+    WHERE ccs.month='2020-05-01' AND ccs.pregnant_all=1 AND awc.state_id='f9b47ea2ee2d8a02acddeeb491d3e175'
```

Comment:
hey this will include those pregnant women as well who are not seeking services. could you use `pregnant=1`  instead of `pregnant_all`. same for lactating below

Reply:
updated